### DISCUSSION TITLE
[v8] Modify cell value before apply a filter

### DISCUSSION THREAD

**Discussion by JuanCruzMoreira** [TanStack/table#5316] [Q&A]
I'm trying to use a filter with date column using a dropdown to select year (año) and month (mes) but the cell value has the complete date. Exist a way to modify the cell value before applying the filter? So I can use getMonth() and getYear() methods or something like that

![image](https://github.com/TanStack/table/assets/94427392/60b063f1-76e8-444a-b70e-fa41a4226f35)

**Comment by goatfryed** [comment]
You can always provide custom filter functions and in there transform the data is required. See the [example on filtering](https://tanstack.com/table/v8/docs/framework/react/examples/filters).

Since you have two independent filter drop downs and you also want the unique options for it (maybe?), it could be better to create two virtual search columns that are not displayed and have a dedicated year and month column just for filtering.

I actually provided feedback to a question about that [here](https://github.com/TanStack/table/discussions/5328) half an hour ago 🙂. Maybe it helps you too.
